### PR TITLE
[TEST] Refactor data lake test suite hierarchical

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Make `*SuiteMixin` resuable.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
Can we also add `DataLakeMixin` to gather duplicates around?

**Comment by pan3793** [comment]
Updated

File: kyuubi-common/src/test/scala/org/apache/kyuubi/DataLakeSuiteMixin.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  protected val catalog: String
+
+  protected val warehouse: Path
+
+  protected val extraJars: String
+
+  protected val extraConfigs: Map[String, String]
+
+  protected val conf: KyuubiConf = {
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/DataLakeSuiteMixin.scala:36]: Not sure if I should put `conf` here

File: kyuubi-common/src/test/scala/org/apache/kyuubi/DataLakeSuiteMixin.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  protected val format: String
+
+  protected val catalog: String
+
+  protected val warehouse: Path
+
+  protected val extraJars: String
+
+  protected val extraConfigs: Map[String, String]
```
Comment (yaooqinn) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/DataLakeSuiteMixin.scala:32]: val -> def

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks for reviewing, merged to master for v1.2.0.
